tests/regression/lp-1764977/task.yaml

```diff
@@ -0,0 +1,22 @@
+summary: Broken snap doesn't break "snap list" and other operations
+details: |
+    Snap can enter the "broken" state when its meta/snap.yaml cannot be found.
+    Typically this means the snap is unmounted or that "snap try"-mode snap was
+    moved on disk and the symlink in /var/lib/snapd/snaps/ is no longer
+    working.
+
+    Broken snaps should not affect snapd operations though, as that forms a
+    denial-of-service attack and may keep machines from refreshing core.
+execute: |
+    cp -a  $TESTSLIB/snaps/test-snapd-sh .
```

Comment:
Maybe a note in tests/main/try-snap-goes-away that there is more tested elsewhere? but maybe overkill.

Reply:
I think we still have to check why that other branch didn't catch this

Reply:
The other test is using `rm -rf` so the entire thing is gone. In the rename case it is still available because it is a bind mount. Just not everything (i.e. the snap symlink is dangling).